app/hermes/views.py

```diff
@@ -21,17 +85,47 @@ async def choose_sales_person(plan: str) -> Admin.pydantic_schema():
         admins = Admin.filter(sells_enterprise=True)
     else:
         raise RequestValidationError('Price plan must be one of "payg,startup,enterprise"')
+
+    region = cf_ipcountry or 'GB'
+
+    if region == 'US':
+        regional_admins = admins.filter(sells_us=True)
+    elif region == 'GB':
+        regional_admins = admins.filter(sells_gb=True)
+    elif region == 'AU':
+        regional_admins = admins.filter(sells_au=True)
+    elif region == 'CA':
+        regional_admins = admins.filter(sells_ca=True)
+    elif region in EU_COUNTRIES:
+        regional_admins = admins.filter(sells_eu=True)
+    else:
+        regional_admins = admins.filter(sells_row=True)
+
     admins = {a.id: a async for a in admins.filter(is_sales_person=True).order_by('id')}
-    admin_ids = list(admins.keys())
+    admins_ids = list(admins.keys())
+
+    regional_admins = {a.id: a async for a in regional_admins.filter(is_sales_person=True).order_by('id')}
+    regional_admins_ids = list(regional_admins.keys())
     latest_company = await Company.filter(price_plan=plan, sales_person_id__isnull=False).order_by('-created').first()
-    if latest_company:
-        latest_sales_person = latest_company.sales_person_id
-        try:
-            next_sales_person = admin_ids[admin_ids.index(latest_sales_person) + 1]
-        except (IndexError, ValueError):
-            next_sales_person = admin_ids[0]
+    if regional_admins:
+        if latest_company:
+            latest_sales_person = latest_company.sales_person_id
+            try:
+                next_sales_person = regional_admins_ids[regional_admins_ids.index(latest_sales_person) + 1]
+            except (IndexError, ValueError):
+                next_sales_person = regional_admins_ids[0]
+        else:
+            next_sales_person = regional_admins_ids[0]
     else:
-        next_sales_person = admin_ids[0]
+        if latest_company:
+            latest_sales_person = latest_company.sales_person_id
+            try:
+                next_sales_person = admins_ids[admins_ids.index(latest_sales_person) + 1]
+            except (IndexError, ValueError):
+                next_sales_person = admins_ids[0]
+        else:
+            next_sales_person = admins_ids[0]
```

Comment:
This function is doing too much now imo and both parts use similar logic, I think for readability you should have a function called like `get_next_sales_person` and then it should take the admins and the latest_company if it exists. So it looks something like 

```python
if regional_admins:
    next_sales_person = _get_next_sales_person(regional_admins)
else:
    next_sales_person = _get_next_sales_person(admins)
```

and then the func could do something like:
```python
def _get_next_sales_person(admins, latest_company):
    if latest_company:
        <your try except code using the admins list passed through>
    else:
        <return like admins[0] or something>
```

Play around with the func tho and make sure it works as thats just pseudo code, then you can add a docstring and typehint so its easier to understand whats going on too.